<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.52">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jeff Walker">
<meta name="dcterms.date" content="2024-06-25">
<meta name="description" content="Two interesting bits with Fig 7d: 1) The researchers analyzed figure 7b data using one-way ANOVA with Tukey adjustment. Post-hoc p-value adjustment is common but typically unjustified for bench biology data like these. The example is a good one to talk about p-value adjustment in bench biology. 2) The linear/ANOVA model implies absurd inference from the CI/SE error bars, including mean counts less than zero!">

<title>Hot take: don’t adjust p-values – Data From</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="../../site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="../../site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Data From</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Hot take: don’t adjust p-values</h1>
            <p class="subtitle lead">Data From Fig 7d – LSD1 inhibition circumvents glucocorticoid-induced muscle wasting of male mice</p>
                  <div>
        <div class="description">
          Two interesting bits with Fig 7d: 1) The researchers analyzed figure 7b data using one-way ANOVA with Tukey adjustment. Post-hoc p-value adjustment is common but typically unjustified for bench biology data like these. The example is a good one to talk about p-value adjustment in bench biology. 2) The linear/ANOVA model implies absurd inference from the CI/SE error bars, including mean counts less than zero!
        </div>
      </div>
                          <div class="quarto-categories">
                <div class="quarto-category">p-values</div>
                <div class="quarto-category">multiple testing</div>
                <div class="quarto-category">generalized linear model</div>
                <div class="quarto-category">absurd CI</div>
                <div class="quarto-category">power</div>
                <div class="quarto-category">simulation</div>
                <div class="quarto-category">pseudoreplication</div>
                <div class="quarto-category">analysis flag</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Jeff Walker </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">June 25, 2024</p>
      </div>
    </div>
    
      <div>
      <div class="quarto-title-meta-heading">Modified</div>
      <div class="quarto-title-meta-contents">
        <p class="date-modified">August 18, 2024</p>
      </div>
    </div>
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#vital-info" id="toc-vital-info" class="nav-link active" data-scroll-target="#vital-info">Vital info</a></li>
  <li><a href="#the-experiment" id="toc-the-experiment" class="nav-link" data-scroll-target="#the-experiment">The Experiment</a></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup">Setup</a></li>
  <li><a href="#import-and-wrangle" id="toc-import-and-wrangle" class="nav-link" data-scroll-target="#import-and-wrangle">Import and Wrangle</a></li>
  <li><a href="#does-the-unstandardization-matter" id="toc-does-the-unstandardization-matter" class="nav-link" data-scroll-target="#does-the-unstandardization-matter">Does the unstandardization matter</a></li>
  <li><a href="#the-reported-p-values-are-conservative-a-bit-on-the-multiple-testing-problem-or-not" id="toc-the-reported-p-values-are-conservative-a-bit-on-the-multiple-testing-problem-or-not" class="nav-link" data-scroll-target="#the-reported-p-values-are-conservative-a-bit-on-the-multiple-testing-problem-or-not">The reported p-values are conservative (a bit on the “multiple testing” problem), or not!</a></li>
  <li><a href="#fig7d-is-a-good-example-of-how-a-linearanova-model-for-count-data-often-implies-silly-confidence-intervals" id="toc-fig7d-is-a-good-example-of-how-a-linearanova-model-for-count-data-often-implies-silly-confidence-intervals" class="nav-link" data-scroll-target="#fig7d-is-a-good-example-of-how-a-linearanova-model-for-count-data-often-implies-silly-confidence-intervals">Fig7d is a good example of how a linear/ANOVA model for count data often implies silly confidence intervals</a></li>
  <li><a href="#a-bit-more-about-generalized-linear-models-glm" id="toc-a-bit-more-about-generalized-linear-models-glm" class="nav-link" data-scroll-target="#a-bit-more-about-generalized-linear-models-glm">A bit more about Generalized Linear Models (GLM)</a></li>
  <li><a href="#for-these-data-the-linearanova-model-has-more-power-than-a-negative-binomial-glm" id="toc-for-these-data-the-linearanova-model-has-more-power-than-a-negative-binomial-glm" class="nav-link" data-scroll-target="#for-these-data-the-linearanova-model-has-more-power-than-a-negative-binomial-glm">For these data, the Linear/ANOVA model has more power than a Negative-Binomial GLM</a></li>
  <li><a href="#lets-plot-the-model-for-all-genes" id="toc-lets-plot-the-model-for-all-genes" class="nav-link" data-scroll-target="#lets-plot-the-model-for-all-genes">Let’s plot the model for all genes</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">





<div class="quarto-figure quarto-figure-left">
<figure class="figure">
<p><img src="../../figs/LSD1 inhibition circumvents glucocorticoid-induced muscle wasting of male mice/fig7d_ggplot.png" class="img-fluid figure-img" style="width:6in"></p>
<figcaption>(A) Reproduced published plot using ANOVA model. Note the modeled CIs imply negative counts and absurd variances. (B) Better-than-reproduced plot using NB-GLM modeled CIs that don’t dip below zero and reflect true variance among groups.</figcaption>
</figure>
</div>
<section id="vital-info" class="level2">
<h2 class="anchored" data-anchor-id="vital-info">Vital info</h2>
<p>Data From: <a href="https://www.nature.com/articles/s41467-024-47846-9" target="_blank">Cai, Q., Sahu, R., Ueberschlag-Pitiot, V., Souali-Crespo, S., Charvet, C., Silem, I., … &amp; Duteil, D. (2024). LSD1 inhibition circumvents glucocorticoid-induced muscle wasting of male mice. Nature Communications, 15(1), 3563.</a></p>
<p>Fig: 7d <a href="../../data from/LSD1 inhibition circumvents glucocorticoid-induced muscle wasting of male mice/41467_2024_47846_MOESM4_ESM.xlsx" target="_blank">download data</a></p>
<p>key words:</p>
<p>Published methods: one-way ANOVA with Tukey adjustment</p>
<p>Design: Completely Randomized Design with subsampling (CRDS)</p>
<p>Response: mRNA level</p>
<p>Key learning concepts: not adjusting for multiple p-values, CIs that go below zero</p>
<p>More info: <a href="https://www.middleprofessor.com/files/applied-biostatistics_bookdown/_book/lmm" target="_blank">Chapter 16 Models for non-independence – linear mixed models</a></p>
</section>
<section id="the-experiment" class="level2">
<h2 class="anchored" data-anchor-id="the-experiment">The Experiment</h2>
<p>Dexamethasone (DEX) is a synthetic glucocortacoid hormone used to treat inflammation and chronic autoimmune diseases but it also stimulates muscle atrophy. The authors are investigating the role of interactions between Lysine-specific demethylase 1 (LSD1) and the glucocortacoid recepter (GCR) in the development of muscle atrophy.</p>
<p>Earlier experiments presented in the paper show evidence that atrophy is mediated by a LSD1/GCR complex that regulates genes related to atrophy and other pathways. For this experiment, mice were treated with a combination of DEX and the LSD1-specific inhibitor CC-90011. Here the researchers are looking at the effect of the different treatment combinations on RNA expression of genes related to the ubiquitin-proteasome system (Fbxo32, Trim63) and the autophagy system (Atg7, Becn1, Bnip3).</p>
<p>Treatment levels</p>
<ol type="1">
<li>Vehicle (Dex-/CC-) Expected to have low levels of target gene expression</li>
<li>Dex (Dex+/CC-) Positive control. Expected to have elevated levels of target gene expression relative to vehicle</li>
<li>CC (Dex-/CC+) Important negative control. We expect this to be similar to Vehicle. If it is not, we need the interaction.</li>
<li>DEX+CC (Dex+/CC+) If LSD1 works with the GCR as expected, then we expect these levels to be lower than Dex+/CC- but how much depends on CC concentration.</li>
</ol>
<p>Analysis Flag: The data are two technical replicates per mouse, so all inference will be a little optimistic. The mouse ID was not archived so I cannot better-than-reproduce the results.</p>
</section>
<section id="setup" class="level2">
<h2 class="anchored" data-anchor-id="setup">Setup</h2>
</section>
<section id="import-and-wrangle" class="level2">
<h2 class="anchored" data-anchor-id="import-and-wrangle">Import and Wrangle</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">"LSD1 inhibition circumvents glucocorticoid-induced muscle wasting of male mice"</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">"41467_2024_47846_MOESM4_ESM.xlsx"</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>read7d <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">range_val =</span> <span class="st">"B3:E13"</span>, <span class="at">label_val =</span> <span class="st">"Fbxo32"</span>){</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  fig7d_part <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">sheet =</span> <span class="st">"Fig. 7d"</span>,</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">range =</span> range_val,</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">col_names =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>() <span class="sc">|&gt;</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">melt</span>(<span class="at">variable.name =</span> <span class="st">"excel_label"</span>, <span class="at">value.name =</span> <span class="st">"y"</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># un-normalize count by multiplying by 10^6</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>  fig7d_part[, y <span class="sc">:</span><span class="er">=</span> <span class="fu">as.integer</span>(y <span class="sc">*</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">6</span>)]</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">setnames</span>(fig7d_part, <span class="at">old =</span> <span class="st">"y"</span>, <span class="at">new =</span> label_val)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  fig7d_part[, mouse_id <span class="sc">:</span><span class="er">=</span> <span class="fu">paste0</span>(<span class="st">"mouse_"</span>, .I)]</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(fig7d_part)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>fig7d <span class="ot">&lt;-</span> <span class="fu">read7d</span>(<span class="at">range_val =</span> <span class="st">"B3:E13"</span>, <span class="at">label_val =</span> <span class="st">"Fbxo32"</span>)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>fig7d <span class="ot">&lt;-</span> <span class="fu">merge</span>(fig7d,</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>               <span class="fu">read7d</span>(<span class="at">range_val =</span> <span class="st">"B16:E26"</span>, <span class="at">label_val =</span> <span class="st">"Trim63"</span>),</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>               <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"excel_label"</span>, <span class="st">"mouse_id"</span>))</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>fig7d <span class="ot">&lt;-</span> <span class="fu">merge</span>(fig7d,</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>               <span class="fu">read7d</span>(<span class="at">range_val =</span> <span class="st">"B29:E39"</span>, <span class="at">label_val =</span> <span class="st">"Atg7"</span>),</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>               <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"excel_label"</span>, <span class="st">"mouse_id"</span>))</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>fig7d <span class="ot">&lt;-</span> <span class="fu">merge</span>(fig7d,</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>               <span class="fu">read7d</span>(<span class="at">range_val =</span> <span class="st">"B42:E52"</span>, <span class="at">label_val =</span> <span class="st">"Becn1"</span>),</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>               <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"excel_label"</span>, <span class="st">"mouse_id"</span>))</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>fig7d <span class="ot">&lt;-</span> <span class="fu">merge</span>(fig7d,</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>               <span class="fu">read7d</span>(<span class="at">range_val =</span> <span class="st">"B55:E65"</span>, <span class="at">label_val =</span> <span class="st">"Bnip3"</span>),</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>               <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"excel_label"</span>, <span class="st">"mouse_id"</span>))</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="co"># remove the missing count</span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>fig7d <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(fig7d)</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="co"># better treatment labels than in excel file</span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>fig7d[excel_label <span class="sc">==</span> <span class="st">"Ctrl Oil"</span>, treatment <span class="sc">:</span><span class="er">=</span> <span class="st">"Vehicle"</span>]</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>fig7d[excel_label <span class="sc">==</span> <span class="st">"DEX"</span>, treatment <span class="sc">:</span><span class="er">=</span> <span class="st">"DEX"</span>]</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>fig7d[excel_label <span class="sc">==</span> <span class="st">"DEX+CC-90011"</span>, treatment <span class="sc">:</span><span class="er">=</span> <span class="st">"DEX+CC"</span>]</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>fig7d[excel_label <span class="sc">==</span> <span class="st">"CC-90011"</span>, treatment <span class="sc">:</span><span class="er">=</span> <span class="st">"CC"</span>]</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>fig7d[, treatment <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(treatment, <span class="at">levels =</span> <span class="fu">unique</span>(fig7d<span class="sc">$</span>treatment))]</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a><span class="co"># split treatment into two factors</span></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>fig7d[treatment <span class="sc">==</span> <span class="st">"Vehicle"</span>, dex <span class="sc">:</span><span class="er">=</span> <span class="st">"Dex_neg"</span>]</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>fig7d[treatment <span class="sc">==</span> <span class="st">"Vehicle"</span>, cc <span class="sc">:</span><span class="er">=</span> <span class="st">"CC_neg"</span>]</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>fig7d[treatment <span class="sc">==</span> <span class="st">"DEX"</span>, dex <span class="sc">:</span><span class="er">=</span> <span class="st">"Dex_pos"</span>]</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>fig7d[treatment <span class="sc">==</span> <span class="st">"DEX"</span>, cc <span class="sc">:</span><span class="er">=</span> <span class="st">"CC_neg"</span>]</span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>fig7d[treatment <span class="sc">==</span> <span class="st">"DEX+CC"</span>, dex <span class="sc">:</span><span class="er">=</span> <span class="st">"Dex_pos"</span>]</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a>fig7d[treatment <span class="sc">==</span> <span class="st">"DEX+CC"</span>, cc <span class="sc">:</span><span class="er">=</span> <span class="st">"CC_pos"</span>]</span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a>fig7d[treatment <span class="sc">==</span> <span class="st">"CC"</span>, dex <span class="sc">:</span><span class="er">=</span> <span class="st">"Dex_neg"</span>]</span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a>fig7d[treatment <span class="sc">==</span> <span class="st">"CC"</span>, cc <span class="sc">:</span><span class="er">=</span> <span class="st">"CC_pos"</span>]</span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a>fig7d[, dex <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(dex, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Dex_neg"</span>, <span class="st">"Dex_pos"</span>))]</span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a>fig7d[, cc <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(cc, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"CC_neg"</span>, <span class="st">"CC_pos"</span>))]</span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a>file_out_name <span class="ot">&lt;-</span> <span class="st">"LSD1 inhibition circumvents glucocorticoid-induced muscle wasting of male mice.xlsx"</span></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a>fileout_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_out_name)</span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a><span class="fu">write_xlsx</span>(fig7d, fileout_path)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="does-the-unstandardization-matter" class="level2">
<h2 class="anchored" data-anchor-id="does-the-unstandardization-matter">Does the unstandardization matter</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>Fbxo32_4 <span class="ot">&lt;-</span> fig7d[, Fbxo32]<span class="sc">/</span><span class="dv">10</span><span class="sc">^</span><span class="dv">6</span> <span class="sc">*</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">4</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(Fbxo32 <span class="sc">~</span> treatment, <span class="at">data =</span> fig7d)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(Fbxo32_4 <span class="sc">~</span> treatment, <span class="at">data =</span> fig7d)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">coef</span>(<span class="fu">summary</span>(m1))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                 Estimate Std. Error   t value     Pr(&gt;|t|)
(Intercept)      999999.5   744928.5 1.3424101 1.881059e-01
treatmentDEX    5499807.1  1053487.9 5.2205695 8.229828e-06
treatmentDEX+CC 2310044.2  1053487.9 2.1927581 3.506200e-02
treatmentCC      127648.4  1082356.0 0.1179357 9.067929e-01</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">coef</span>(<span class="fu">summary</span>(m2))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                 Estimate Std. Error   t value     Pr(&gt;|t|)
(Intercept)      9999.995   7449.285 1.3424101 1.881059e-01
treatmentDEX    54998.071  10534.879 5.2205695 8.229828e-06
treatmentDEX+CC 23100.442  10534.879 2.1927581 3.506200e-02
treatmentCC      1276.484  10823.560 0.1179357 9.067929e-01</code></pre>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(Fbxo32 <span class="sc">~</span> treatment, <span class="at">data =</span> fig7d)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="the-reported-p-values-are-conservative-a-bit-on-the-multiple-testing-problem-or-not" class="level2">
<h2 class="anchored" data-anchor-id="the-reported-p-values-are-conservative-a-bit-on-the-multiple-testing-problem-or-not">The reported p-values are conservative (a bit on the “multiple testing” problem), or not!</h2>
<p>The researchers reported Tukey-adjusted p-values. The Tukey method adjusts for the number of tests in a “family” of tests, where a family is the set of tests that answer one question. The researchers follow the norm in bench bio and choose the whole set of six contrasts as the family. This would answer a question such as: “if we were to create these combinations, would we find any difference between at least on pair?”, which isn’t a very interesting question.</p>
<p>The researchers are focused on a much more specific question and at least two more questions, both related to controls:</p>
<ol type="1">
<li>Family 1: Is there evidence that CC attenuates DEX-induced expression? There is only one test in this family: DEX+CC - DEXI. This is the focal family.</li>
<li>Family 2: Is there evidence that DEX increase expression levels, as expected? This is the “positive control” family. There is only one test in this family: DEX - Vehicle.</li>
<li>Family 3: What is the effect of CC alone? This is “negative control” family. Without DEX-induced elevated express, we would expect this response to be similar to Vehicle. There is only one test in this family: CC - Vehicle.</li>
</ol>
<p>What about the interaction effect? The contrast (DEX+CC - DEX) is the focal contrast <em>if there is no effect of CC alone</em>. We could look at the CC alone effect (CC - Vehicle) and if it is sufficiently small, use (DEX+CC - DEX) as the focal contrast. Of course “sufficiently small” has a subjective bright line. The statistically correct way to address this is to compare the effect of CC with DEX (the focal contrast) to the effect of CC without DEX (CC - Vehicle). This comparison is the interaction effect. So we’ll add the interaction contrast as a 4th family, which we should think of as replacing the focal contrast (Family 1) and not “in addition to” the three families. But, I’ll compute both. Regardless, for these data, I think most researchers would agree that CC - Vehicle is sufficiently small that we can use the DEX+CC - DEX comparison without issue.</p>
<p>So we have three p-values that we care about and each is from a different family, so there is no justification for p-value adjustment for multiple tests. This would make the reported p-values conservative. In general, if I wanted to be more conservative, in order to avoid research time and waste, I’d simply make my “alpha” smaller, such as 0.01 or 0.005 instead of using more conservative tests like a Tukey p-value adjustment, since the conservative alpha doesn’t depend on things like number of tests in a family.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># the researcher's analysis</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>m0 <span class="ot">&lt;-</span> <span class="fu">lm</span>(Fbxo32 <span class="sc">~</span> treatment, <span class="at">data =</span> fig7d)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>m0_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(m0, <span class="fu">c</span>(<span class="st">"treatment"</span>))</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>m0_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(m0_emm,</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">"revpairwise"</span>,</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">adjust =</span> <span class="st">"tukey"</span>) <span class="sc">|&gt;</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co"># the analysis only comparing the families of tests that we care about</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>veh <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>dex <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>)</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>dex_cc <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>) <span class="co"># note the wonky ordering in fig 7d</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>cc <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>focal_contrasts <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">"DEX - Vehicle"</span> <span class="ot">=</span> dex <span class="sc">-</span> veh,</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  <span class="st">"CC - Vehicle"</span> <span class="ot">=</span> cc <span class="sc">-</span> veh,</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">"DEX+CC - DEX"</span> <span class="ot">=</span> dex_cc <span class="sc">-</span> dex,</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">"interaction"</span> <span class="ot">=</span> (dex_cc <span class="sc">-</span> dex) <span class="sc">-</span> (cc <span class="sc">-</span> veh)</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(Fbxo32 <span class="sc">~</span> treatment, <span class="at">data =</span> fig7d)</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(m1, <span class="fu">c</span>(<span class="st">"treatment"</span>))</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>m1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(m1_emm,</span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> focal_contrasts,</span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>                     <span class="at">adjust =</span> <span class="st">"none"</span>) <span class="sc">|&gt;</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>contrast_table <span class="ot">&lt;-</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(m0_pairs, m1_pairs)</span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>contrast_table <span class="sc">|&gt;</span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">4</span>) <span class="sc">|&gt;</span></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">|&gt;</span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pack_rows</span>(<span class="st">"Adjusted"</span>, <span class="dv">1</span>, <span class="dv">6</span>) <span class="sc">|&gt;</span></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pack_rows</span>(<span class="st">"Non-adjusted"</span>, <span class="dv">7</span>, <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="6">
<td colspan="8" style="border-bottom: 1px solid"><strong>Adjusted</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">DEX - Vehicle</td>
<td style="text-align: right;">5499807.1</td>
<td style="text-align: right;">1053488</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">2658652.5</td>
<td style="text-align: right;">8340961.7</td>
<td style="text-align: right;">5.2206</td>
<td style="text-align: right;">0.0000</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(DEX+CC) - Vehicle</td>
<td style="text-align: right;">2310044.2</td>
<td style="text-align: right;">1053488</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">-531110.4</td>
<td style="text-align: right;">5151198.8</td>
<td style="text-align: right;">2.1928</td>
<td style="text-align: right;">0.1452</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(DEX+CC) - DEX</td>
<td style="text-align: right;">-3189762.9</td>
<td style="text-align: right;">1053488</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">-6030917.5</td>
<td style="text-align: right;">-348608.3</td>
<td style="text-align: right;">-3.0278</td>
<td style="text-align: right;">0.0228</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">CC - Vehicle</td>
<td style="text-align: right;">127648.4</td>
<td style="text-align: right;">1082356</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">-2791360.5</td>
<td style="text-align: right;">3046657.2</td>
<td style="text-align: right;">0.1179</td>
<td style="text-align: right;">0.9994</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">CC - DEX</td>
<td style="text-align: right;">-5372158.7</td>
<td style="text-align: right;">1082356</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">-8291167.6</td>
<td style="text-align: right;">-2453149.9</td>
<td style="text-align: right;">-4.9634</td>
<td style="text-align: right;">0.0001</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">CC - (DEX+CC)</td>
<td style="text-align: right;">-2182395.8</td>
<td style="text-align: right;">1082356</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">-5101404.7</td>
<td style="text-align: right;">736613.0</td>
<td style="text-align: right;">-2.0163</td>
<td style="text-align: right;">0.2015</td>
</tr>
<tr class="even" data-grouplength="4">
<td colspan="8" style="border-bottom: 1px solid"><strong>Non-adjusted</strong></td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">DEX - Vehicle</td>
<td style="text-align: right;">5499807.1</td>
<td style="text-align: right;">1053488</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">3361112.9</td>
<td style="text-align: right;">7638501.3</td>
<td style="text-align: right;">5.2206</td>
<td style="text-align: right;">0.0000</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">CC - Vehicle</td>
<td style="text-align: right;">127648.4</td>
<td style="text-align: right;">1082356</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">-2069651.1</td>
<td style="text-align: right;">2324947.8</td>
<td style="text-align: right;">0.1179</td>
<td style="text-align: right;">0.9068</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">DEX+CC - DEX</td>
<td style="text-align: right;">-3189762.9</td>
<td style="text-align: right;">1053488</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">-5328457.1</td>
<td style="text-align: right;">-1051068.7</td>
<td style="text-align: right;">-3.0278</td>
<td style="text-align: right;">0.0046</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">interaction</td>
<td style="text-align: right;">-3317411.3</td>
<td style="text-align: right;">1510408</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">-6383701.9</td>
<td style="text-align: right;">-251120.7</td>
<td style="text-align: right;">-2.1964</td>
<td style="text-align: right;">0.0348</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>The top table contains the six Tukey-adjusted p-values reported by the researchers. The bottom table contains the unadjusted values of only the contrasts of interest. The adjusted <em>p</em>-value for the focal contrast DEX+CC - DEX is 0.023 while the unadjusted value is 0.0046.</p>
<p>The p-value of the interaction is 0.035 – note that it’s effect size is very similar to that of the DEX+CC - DEX but the p-value is bigger because the SE of an interaction is bigger than the SE of a main contrast (because the interaction is a function of 4 means and not just 2). Using the interaction p-value is not just statistically correct, it is more conservative.</p>
</section>
<section id="fig7d-is-a-good-example-of-how-a-linearanova-model-for-count-data-often-implies-silly-confidence-intervals" class="level2">
<h2 class="anchored" data-anchor-id="fig7d-is-a-good-example-of-how-a-linearanova-model-for-count-data-often-implies-silly-confidence-intervals">Fig7d is a good example of how a linear/ANOVA model for count data often implies silly confidence intervals</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>gg1 <span class="ot">&lt;-</span> <span class="fu">plot_response</span>(m1,</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>              m1_emm,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>              m1_pairs[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,],</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">palette =</span> <span class="st">"pal_okabe_ito_blue"</span>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">glm.nb</span>(Fbxo32 <span class="sc">~</span> treatment, <span class="at">data =</span> fig7d)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>m2_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(m2, <span class="fu">c</span>(<span class="st">"treatment"</span>), <span class="at">type =</span> <span class="st">"response"</span>)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>m2_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(m2_emm,</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> focal_contrasts,</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>                     <span class="at">adjust =</span> <span class="st">"none"</span>) <span class="sc">|&gt;</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>m2_pairs <span class="ot">&lt;-</span> m2_pairs[, <span class="sc">-</span><span class="dv">7</span>]</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> <span class="fu">plot_response</span>(m2,</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>              m2_emm,</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>              m2_pairs[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,],</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">palette =</span> <span class="st">"pal_okabe_ito_blue"</span>)</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(gg1, gg2, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">labels =</span> <span class="st">"AUTO"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="LSD1-inhibition-circumvents-glucocorticoid-induced-muscle-wasting-of-male-mice_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>save_it <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(save_it){</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  out_fig <span class="ot">&lt;-</span> <span class="st">"fig7d_ggplot.png"</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  out_path <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"figs"</span>, data_from, out_fig)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(out_path)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The two plots above are the estimated means and 95% CIs from the linear/ANOVA model (A) and a negative-binomial Generalized Linear Model (B). The linear/ANOVA model implies two absurd things about the confidence of the means of the groups</p>
<ol type="1">
<li>Homogeneity of variances – remember, equal variance among treatment levels is an assumption of a linear/ANOVA model, so the model standard error of the mean of a group is a function only of the group sample size. The modeled SEs/CIs are too big for the groups with small mean counts and too small for the groups with large mean counts.</li>
<li>Values below zero. A good way to think about a CI is, values within a confidence interval are reasonably plausible values for the group mean. The CIs of Vehicle and CC imply that negative means are “reasonably plausible” values, which is absurd, counts cannot be negative.</li>
</ol>
<p>By contrast, the CIs from the GLM more accurately model the data.</p>
<ol type="1">
<li>The modeled CIs vary among the groups and look much more like the sampled CIs. In a negative-binomial GLM, the model standard error of the mean of a group is a function of the sample size and the mean of the group – groups with larger means have larger variances/standard errors. It’s easy to see this phenomenon with these data; the spread of points for DEX is much more than that for Vehicle, but the linear/ANOVA model CIs do not model this.</li>
<li>the CIs are asymmetric, the upper bound is further from the mean than the lower bound. Again, this reflects the skew of the data.</li>
<li>the CIs do not (and cannot) dip below zero.</li>
</ol>
<p>We don’t see figures like Panel A above because researchers plot <strong>sample</strong> SEs and not modeled SEs – the SE actually used to compute the <em>p</em>-value. So “inference from the plot” is inconsistent with the <em>p</em>-value. This is why researchers should plot the model (modeled SE/CI) and not the data (sampled SE/CI).</p>
</section>
<section id="a-bit-more-about-generalized-linear-models-glm" class="level2">
<h2 class="anchored" data-anchor-id="a-bit-more-about-generalized-linear-models-glm">A bit more about Generalized Linear Models (GLM)</h2>
<p>Count data in biology often have a distribution that nicely approximates a sample from a Negative Binomial or a Quasipoisson distribution. Two features of count data that violate assumptions of a linear/ANOVA model are</p>
<ol type="1">
<li>Homeogeneity of variance – the variance of the response is proportional to the mean of the group, so groups with a higher mean count have higher variances</li>
<li>Normal – counts are right skewed, that is, there is a high density of low counts and a low density of high counts</li>
</ol>
<p>Generalized Linear Models (GLM) were developed to compute confidence intervals and <em>p</em>-values for counts and other kinds of data with non-normal distributions.</p>
</section>
<section id="for-these-data-the-linearanova-model-has-more-power-than-a-negative-binomial-glm" class="level2">
<h2 class="anchored" data-anchor-id="for-these-data-the-linearanova-model-has-more-power-than-a-negative-binomial-glm">For these data, the Linear/ANOVA model has more power than a Negative-Binomial GLM</h2>
<p>Despite the violations, tests from linear/ANOVA models that give a <em>p</em>-value often have good long-run performance, that is, they have relatively high power and controlled type I error <a href="https://besjournals.onlinelibrary.wiley.com/doi/abs/10.1111/2041-210X.12386" target="_blank">Ives 2015</a>. GLM models for count data can have higher power than a linear/ANOVA model, but this sometimes requires some more sophisticated methods such as bootstrap resampling <a href="https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.12552" target="_blank">Warton et al.&nbsp;2016</a>.</p>
<p>Here I use a simulation to compare the Type I error and Power of the 1) Linear/ANOVA model, 2) a Negative Binomial GLM, and 3) a quasipoisson GLM for data that look like those in Figure 7d. I explicitly simulate the Trim63 response with the same unbalanced data as in the experiment (the CC group only has 9 measures and not 10).</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get parameters</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>do_it <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(do_it){</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  data_params <span class="ot">&lt;-</span> fig7d[, .(<span class="at">mean =</span> <span class="fu">mean</span>(Trim63),</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">N =</span> .N), by <span class="ot">=</span> <span class="st">"treatment"</span>]</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  mu <span class="ot">&lt;-</span> <span class="fu">rep</span>(data_params<span class="sc">$</span>mean, data_params<span class="sc">$</span>N)</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  N <span class="ot">&lt;-</span> <span class="fu">sum</span>(data_params<span class="sc">$</span>N)</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  m1 <span class="ot">&lt;-</span> <span class="fu">glm.nb</span>(Trim63 <span class="sc">~</span> dex <span class="sc">*</span> cc, <span class="at">data =</span> fig7d)</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  theta_sim <span class="ot">&lt;-</span> m1<span class="sc">$</span>theta</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>  n_sim <span class="ot">&lt;-</span> <span class="dv">2000</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>  fd <span class="ot">&lt;-</span> <span class="fu">data.table</span>(</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">treatment =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"veh"</span>, <span class="st">"dex"</span>, <span class="st">"dex_cc"</span>, <span class="st">"cc"</span>), data_params<span class="sc">$</span>N),</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">dex =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"no_dex"</span>, <span class="st">"dex"</span>, <span class="st">"dex"</span>, <span class="st">"no_dex"</span>), data_params<span class="sc">$</span>N),</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">cc =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"no_cc"</span>, <span class="st">"no_cc"</span>, <span class="st">"cc"</span>, <span class="st">"cc"</span>), data_params<span class="sc">$</span>N),</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">Trim63 =</span> <span class="fu">as.numeric</span>(<span class="cn">NA</span>)</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># power</span></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>  power_matrix <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="at">nrow =</span> n_sim, <span class="at">ncol =</span> <span class="dv">6</span>)</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(power_matrix) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"lm_focal"</span>, <span class="st">"lm_ixn"</span>, <span class="st">"nb_focal"</span>, <span class="st">"nb_ixn"</span>, <span class="st">"qp_focal"</span>, <span class="st">"qp_ixn"</span>)</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(sim_i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_sim){</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># fake data</span></span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>    fd[, Trim63 <span class="sc">:</span><span class="er">=</span> <span class="fu">rnegbin</span>(N, mu, theta_sim)]</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># lm/ANOVA</span></span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>    m_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(Trim63 <span class="sc">~</span> dex <span class="sc">*</span> cc, <span class="at">data =</span> fd)</span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a>    power_matrix[sim_i, <span class="st">"lm_ixn"</span>] <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(m_lm))[<span class="st">"dexno_dex:ccno_cc"</span>, <span class="st">"Pr(&gt;|t|)"</span>]</span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a>    m_lm_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(m_lm, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"dex"</span>, <span class="st">"cc"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a>      <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">"revpairwise"</span>, <span class="at">simple =</span> <span class="st">"each"</span>, <span class="at">combine =</span> <span class="cn">TRUE</span>, <span class="at">adjust =</span> <span class="st">"none"</span>) <span class="sc">|&gt;</span></span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a>    power_matrix[sim_i, <span class="st">"lm_focal"</span>] <span class="ot">&lt;-</span> m_lm_pairs[<span class="dv">3</span>, <span class="st">"p.value"</span>] <span class="co"># effect of CC when dex is present</span></span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a>    <span class="co"># glm neg bin</span></span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a>    m_nb <span class="ot">&lt;-</span> <span class="fu">glm.nb</span>(Trim63 <span class="sc">~</span> dex <span class="sc">*</span> cc, <span class="at">data =</span> fd)</span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a>    power_matrix[sim_i, <span class="st">"nb_ixn"</span>] <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(m_nb))[<span class="st">"dexno_dex:ccno_cc"</span>, <span class="st">"Pr(&gt;|z|)"</span>]</span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a>    m_nb_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(m_nb, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"dex"</span>, <span class="st">"cc"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a>      <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">"revpairwise"</span>, <span class="at">simple =</span> <span class="st">"each"</span>, <span class="at">combine =</span> <span class="cn">TRUE</span>, <span class="at">adjust =</span> <span class="st">"none"</span>) <span class="sc">|&gt;</span></span>
<span id="cb10-39"><a href="#cb10-39" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-40"><a href="#cb10-40" aria-hidden="true" tabindex="-1"></a>    power_matrix[sim_i, <span class="st">"nb_focal"</span>] <span class="ot">&lt;-</span> m_nb_pairs[<span class="dv">3</span>, <span class="st">"p.value"</span>] <span class="co"># effect of CC when dex is present</span></span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a>    <span class="co"># glm quasipoisson</span></span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a>    m_qp <span class="ot">&lt;-</span> <span class="fu">glm</span>(Trim63 <span class="sc">~</span> dex <span class="sc">*</span> cc,</span>
<span id="cb10-44"><a href="#cb10-44" aria-hidden="true" tabindex="-1"></a>                <span class="at">family =</span> <span class="fu">quasipoisson</span>(<span class="at">link =</span> <span class="st">"log"</span>),</span>
<span id="cb10-45"><a href="#cb10-45" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> fd)</span>
<span id="cb10-46"><a href="#cb10-46" aria-hidden="true" tabindex="-1"></a>    power_matrix[sim_i, <span class="st">"qp_ixn"</span>] <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(m_qp))[<span class="st">"dexno_dex:ccno_cc"</span>, <span class="st">"Pr(&gt;|t|)"</span>]</span>
<span id="cb10-47"><a href="#cb10-47" aria-hidden="true" tabindex="-1"></a>    m_qp_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(m_qp, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"dex"</span>, <span class="st">"cc"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb10-48"><a href="#cb10-48" aria-hidden="true" tabindex="-1"></a>      <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">"revpairwise"</span>, <span class="at">simple =</span> <span class="st">"each"</span>, <span class="at">combine =</span> <span class="cn">TRUE</span>, <span class="at">adjust =</span> <span class="st">"none"</span>) <span class="sc">|&gt;</span></span>
<span id="cb10-49"><a href="#cb10-49" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-50"><a href="#cb10-50" aria-hidden="true" tabindex="-1"></a>    power_matrix[sim_i, <span class="st">"qp_focal"</span>] <span class="ot">&lt;-</span> m_qp_pairs[<span class="dv">3</span>, <span class="st">"p.value"</span>] <span class="co"># effect of CC when dex is present</span></span>
<span id="cb10-51"><a href="#cb10-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-52"><a href="#cb10-52" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb10-53"><a href="#cb10-53" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-54"><a href="#cb10-54" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-55"><a href="#cb10-55" aria-hidden="true" tabindex="-1"></a>  <span class="co"># type I</span></span>
<span id="cb10-56"><a href="#cb10-56" aria-hidden="true" tabindex="-1"></a>  type1_matrix <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="at">nrow =</span> n_sim, <span class="at">ncol =</span> <span class="dv">6</span>)</span>
<span id="cb10-57"><a href="#cb10-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(type1_matrix) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"lm_focal"</span>, <span class="st">"lm_ixn"</span>, <span class="st">"nb_focal"</span>, <span class="st">"nb_ixn"</span>, <span class="st">"qp_focal"</span>, <span class="st">"qp_ixn"</span>)</span>
<span id="cb10-58"><a href="#cb10-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-59"><a href="#cb10-59" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(sim_i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_sim){</span>
<span id="cb10-60"><a href="#cb10-60" aria-hidden="true" tabindex="-1"></a>    <span class="co"># fake data</span></span>
<span id="cb10-61"><a href="#cb10-61" aria-hidden="true" tabindex="-1"></a>    fd[, Trim63 <span class="sc">:</span><span class="er">=</span> <span class="fu">rnegbin</span>(N, mu[<span class="dv">1</span>], theta_sim)]</span>
<span id="cb10-62"><a href="#cb10-62" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-63"><a href="#cb10-63" aria-hidden="true" tabindex="-1"></a>    <span class="co"># lm/ANOVA</span></span>
<span id="cb10-64"><a href="#cb10-64" aria-hidden="true" tabindex="-1"></a>    m_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(Trim63 <span class="sc">~</span> dex <span class="sc">*</span> cc, <span class="at">data =</span> fd)</span>
<span id="cb10-65"><a href="#cb10-65" aria-hidden="true" tabindex="-1"></a>    type1_matrix[sim_i, <span class="st">"lm_ixn"</span>] <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(m_lm))[<span class="st">"dexno_dex:ccno_cc"</span>, <span class="st">"Pr(&gt;|t|)"</span>]</span>
<span id="cb10-66"><a href="#cb10-66" aria-hidden="true" tabindex="-1"></a>    m_lm_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(m_lm, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"dex"</span>, <span class="st">"cc"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb10-67"><a href="#cb10-67" aria-hidden="true" tabindex="-1"></a>      <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">"revpairwise"</span>, <span class="at">simple =</span> <span class="st">"each"</span>, <span class="at">combine =</span> <span class="cn">TRUE</span>, <span class="at">adjust =</span> <span class="st">"none"</span>) <span class="sc">|&gt;</span></span>
<span id="cb10-68"><a href="#cb10-68" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-69"><a href="#cb10-69" aria-hidden="true" tabindex="-1"></a>    type1_matrix[sim_i, <span class="st">"lm_focal"</span>] <span class="ot">&lt;-</span> m_lm_pairs[<span class="dv">3</span>, <span class="st">"p.value"</span>] <span class="co"># effect of CC when dex is present</span></span>
<span id="cb10-70"><a href="#cb10-70" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-71"><a href="#cb10-71" aria-hidden="true" tabindex="-1"></a>    <span class="co"># glm neg bin</span></span>
<span id="cb10-72"><a href="#cb10-72" aria-hidden="true" tabindex="-1"></a>    m_nb <span class="ot">&lt;-</span> <span class="fu">glm.nb</span>(Trim63 <span class="sc">~</span> dex <span class="sc">*</span> cc, <span class="at">data =</span> fd)</span>
<span id="cb10-73"><a href="#cb10-73" aria-hidden="true" tabindex="-1"></a>    type1_matrix[sim_i, <span class="st">"nb_ixn"</span>] <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(m_nb))[<span class="st">"dexno_dex:ccno_cc"</span>, <span class="st">"Pr(&gt;|z|)"</span>]</span>
<span id="cb10-74"><a href="#cb10-74" aria-hidden="true" tabindex="-1"></a>    m_nb_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(m_nb, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"dex"</span>, <span class="st">"cc"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb10-75"><a href="#cb10-75" aria-hidden="true" tabindex="-1"></a>      <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">"revpairwise"</span>, <span class="at">simple =</span> <span class="st">"each"</span>, <span class="at">combine =</span> <span class="cn">TRUE</span>, <span class="at">adjust =</span> <span class="st">"none"</span>) <span class="sc">|&gt;</span></span>
<span id="cb10-76"><a href="#cb10-76" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-77"><a href="#cb10-77" aria-hidden="true" tabindex="-1"></a>    type1_matrix[sim_i, <span class="st">"nb_focal"</span>] <span class="ot">&lt;-</span> m_nb_pairs[<span class="dv">3</span>, <span class="st">"p.value"</span>] <span class="co"># effect of CC when dex is present</span></span>
<span id="cb10-78"><a href="#cb10-78" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-79"><a href="#cb10-79" aria-hidden="true" tabindex="-1"></a>    <span class="co"># glm quasipoisson</span></span>
<span id="cb10-80"><a href="#cb10-80" aria-hidden="true" tabindex="-1"></a>    m_qp <span class="ot">&lt;-</span> <span class="fu">glm</span>(Trim63 <span class="sc">~</span> dex <span class="sc">*</span> cc,</span>
<span id="cb10-81"><a href="#cb10-81" aria-hidden="true" tabindex="-1"></a>                <span class="at">family =</span> <span class="fu">quasipoisson</span>(<span class="at">link =</span> <span class="st">"log"</span>),</span>
<span id="cb10-82"><a href="#cb10-82" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> fd)</span>
<span id="cb10-83"><a href="#cb10-83" aria-hidden="true" tabindex="-1"></a>    type1_matrix[sim_i, <span class="st">"qp_ixn"</span>] <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(m_qp))[<span class="st">"dexno_dex:ccno_cc"</span>, <span class="st">"Pr(&gt;|t|)"</span>]</span>
<span id="cb10-84"><a href="#cb10-84" aria-hidden="true" tabindex="-1"></a>    m_qp_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(m_qp, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"dex"</span>, <span class="st">"cc"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb10-85"><a href="#cb10-85" aria-hidden="true" tabindex="-1"></a>      <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">"revpairwise"</span>, <span class="at">simple =</span> <span class="st">"each"</span>, <span class="at">combine =</span> <span class="cn">TRUE</span>, <span class="at">adjust =</span> <span class="st">"none"</span>) <span class="sc">|&gt;</span></span>
<span id="cb10-86"><a href="#cb10-86" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-87"><a href="#cb10-87" aria-hidden="true" tabindex="-1"></a>    type1_matrix[sim_i, <span class="st">"qp_focal"</span>] <span class="ot">&lt;-</span> m_qp_pairs[<span class="dv">3</span>, <span class="st">"p.value"</span>] <span class="co"># effect of CC when dex is present</span></span>
<span id="cb10-88"><a href="#cb10-88" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb10-89"><a href="#cb10-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(power_matrix, <span class="st">"power_matrix.Rds"</span>)</span>
<span id="cb10-90"><a href="#cb10-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(type1_matrix, <span class="st">"type1_matrix.Rds"</span>)</span>
<span id="cb10-91"><a href="#cb10-91" aria-hidden="true" tabindex="-1"></a>}<span class="cf">else</span>{</span>
<span id="cb10-92"><a href="#cb10-92" aria-hidden="true" tabindex="-1"></a>  power_matrix <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"power_matrix.Rds"</span>)</span>
<span id="cb10-93"><a href="#cb10-93" aria-hidden="true" tabindex="-1"></a>  type1_matrix <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"type1_matrix.Rds"</span>)</span>
<span id="cb10-94"><a href="#cb10-94" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-95"><a href="#cb10-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-96"><a href="#cb10-96" aria-hidden="true" tabindex="-1"></a>pless <span class="ot">&lt;-</span> <span class="cf">function</span>(x){</span>
<span id="cb10-97"><a href="#cb10-97" aria-hidden="true" tabindex="-1"></a>  stat <span class="ot">&lt;-</span> <span class="fu">sum</span>(x <span class="sc">&lt;</span> <span class="fl">0.05</span>)<span class="sc">/</span><span class="fu">length</span>(x)</span>
<span id="cb10-98"><a href="#cb10-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(stat)</span>
<span id="cb10-99"><a href="#cb10-99" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-100"><a href="#cb10-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-101"><a href="#cb10-101" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">data.table</span>(</span>
<span id="cb10-102"><a href="#cb10-102" aria-hidden="true" tabindex="-1"></a>  <span class="at">Test =</span> <span class="fu">c</span>(<span class="st">"LM: DEX+CC - DEX"</span>, <span class="st">"LM: Interaction"</span>,</span>
<span id="cb10-103"><a href="#cb10-103" aria-hidden="true" tabindex="-1"></a>           <span class="st">"GLM-NB: DEX+CC - DEX"</span>, <span class="st">"GLM-NB: Interaction"</span>,</span>
<span id="cb10-104"><a href="#cb10-104" aria-hidden="true" tabindex="-1"></a>           <span class="st">"GLM-QP: DEX+CC - DEX"</span>, <span class="st">"GLM-QP: Interaction"</span>),</span>
<span id="cb10-105"><a href="#cb10-105" aria-hidden="true" tabindex="-1"></a>  <span class="at">Power =</span> <span class="fu">apply</span>(power_matrix, <span class="dv">2</span>, pless),</span>
<span id="cb10-106"><a href="#cb10-106" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Type I"</span> <span class="ot">=</span> <span class="fu">apply</span>(type1_matrix, <span class="dv">2</span>, pless)</span>
<span id="cb10-107"><a href="#cb10-107" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-108"><a href="#cb10-108" aria-hidden="true" tabindex="-1"></a>res <span class="sc">|&gt;</span></span>
<span id="cb10-109"><a href="#cb10-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">|&gt;</span></span>
<span id="cb10-110"><a href="#cb10-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Test</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Power</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Type I</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">LM: DEX+CC - DEX</td>
<td style="text-align: right;">0.8120</td>
<td style="text-align: right;">0.0520</td>
</tr>
<tr class="even">
<td style="text-align: left;">LM: Interaction</td>
<td style="text-align: right;">0.6930</td>
<td style="text-align: right;">0.0475</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GLM-NB: DEX+CC - DEX</td>
<td style="text-align: right;">0.7340</td>
<td style="text-align: right;">0.0745</td>
</tr>
<tr class="even">
<td style="text-align: left;">GLM-NB: Interaction</td>
<td style="text-align: right;">0.6970</td>
<td style="text-align: right;">0.0685</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GLM-QP: DEX+CC - DEX</td>
<td style="text-align: right;">0.7935</td>
<td style="text-align: right;">0.0575</td>
</tr>
<tr class="even">
<td style="text-align: left;">GLM-QP: Interaction</td>
<td style="text-align: right;">0.5595</td>
<td style="text-align: right;">0.0490</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Curious. The linear model with an untransformed response has better Type I control and slightly more power than a negative binomial GLM.</p>
<p>That said, I really like the GLM for two reasons:</p>
<ol type="1">
<li>The effect is a multiple of the reference, which is a really meaningful way of comparing responses (for example, looking at the table below, the effect is 6.5, meaning the expression level is 6.5 times larger than that of Vehicle)</li>
<li>The CIs of the means are asymmetric, which reflects the skewed distribution of the data. This also avoids absurd CIs such as a negative lower bound (counts can’t be negative!).</li>
</ol>
</section>
<section id="lets-plot-the-model-for-all-genes" class="level2">
<h2 class="anchored" data-anchor-id="lets-plot-the-model-for-all-genes">Let’s plot the model for all genes</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>get_model_result <span class="ot">&lt;-</span> <span class="cf">function</span>(gene_i){</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  model_formula <span class="ot">&lt;-</span> <span class="fu">paste</span>(gene_i, <span class="st">"~ treatment"</span>) <span class="sc">|&gt;</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>()</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  m1 <span class="ot">&lt;-</span> <span class="fu">glm.nb</span>(model_formula, <span class="at">data =</span> fig7d)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"treatment"</span>), <span class="at">type =</span> <span class="st">"response"</span>)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  m1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(m1_emm,</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> focal_contrasts,</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">"none"</span>) <span class="sc">|&gt;</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">m1 =</span> m1, <span class="at">m1_emm =</span> m1_emm, <span class="at">m1_pairs =</span> m1_pairs))</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>res1 <span class="ot">&lt;-</span> <span class="fu">get_model_result</span>(<span class="st">"Fbxo32"</span>)</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>gg1 <span class="ot">&lt;-</span> <span class="fu">plot_response</span>(res1<span class="sc">$</span>m1, res1<span class="sc">$</span>m1_emm, res1<span class="sc">$</span>m1_pairs[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>),], <span class="at">palette =</span> <span class="st">"pal_okabe_ito_blue"</span>,</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>                     <span class="at">x_axis_labels =</span> <span class="fu">c</span>(<span class="st">"DEX-</span><span class="sc">\n</span><span class="st">CC-"</span>, <span class="st">"DEX+</span><span class="sc">\n</span><span class="st">CC-"</span>, <span class="st">"DEX+</span><span class="sc">\n</span><span class="st">CC+"</span>, <span class="st">"DEX-</span><span class="sc">\n</span><span class="st">CC+"</span>))</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>res2 <span class="ot">&lt;-</span> <span class="fu">get_model_result</span>(<span class="st">"Trim63"</span>)</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> <span class="fu">plot_response</span>(res2<span class="sc">$</span>m1, res2<span class="sc">$</span>m1_emm, res2<span class="sc">$</span>m1_pairs[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>),], <span class="at">palette =</span> <span class="st">"pal_okabe_ito_blue"</span>,</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>                     <span class="at">x_axis_labels =</span> <span class="fu">c</span>(<span class="st">"DEX-</span><span class="sc">\n</span><span class="st">CC-"</span>, <span class="st">"DEX+</span><span class="sc">\n</span><span class="st">CC-"</span>, <span class="st">"DEX+</span><span class="sc">\n</span><span class="st">CC+"</span>, <span class="st">"DEX-</span><span class="sc">\n</span><span class="st">CC+"</span>))</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>res3 <span class="ot">&lt;-</span> <span class="fu">get_model_result</span>(<span class="st">"Atg7"</span>)</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>gg3 <span class="ot">&lt;-</span> <span class="fu">plot_response</span>(res3<span class="sc">$</span>m1, res3<span class="sc">$</span>m1_emm, res3<span class="sc">$</span>m1_pairs[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>),], <span class="at">palette =</span> <span class="st">"pal_okabe_ito_blue"</span>,</span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>                     <span class="at">x_axis_labels =</span> <span class="fu">c</span>(<span class="st">"DEX-</span><span class="sc">\n</span><span class="st">CC-"</span>, <span class="st">"DEX+</span><span class="sc">\n</span><span class="st">CC-"</span>, <span class="st">"DEX+</span><span class="sc">\n</span><span class="st">CC+"</span>, <span class="st">"DEX-</span><span class="sc">\n</span><span class="st">CC+"</span>))</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>res4 <span class="ot">&lt;-</span> <span class="fu">get_model_result</span>(<span class="st">"Becn1"</span>)</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>gg4 <span class="ot">&lt;-</span> <span class="fu">plot_response</span>(res4<span class="sc">$</span>m1, res4<span class="sc">$</span>m1_emm, res4<span class="sc">$</span>m1_pairs[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>),], <span class="at">palette =</span> <span class="st">"pal_okabe_ito_blue"</span>,</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>                     <span class="at">x_axis_labels =</span> <span class="fu">c</span>(<span class="st">"DEX-</span><span class="sc">\n</span><span class="st">CC-"</span>, <span class="st">"DEX+</span><span class="sc">\n</span><span class="st">CC-"</span>, <span class="st">"DEX+</span><span class="sc">\n</span><span class="st">CC+"</span>, <span class="st">"DEX-</span><span class="sc">\n</span><span class="st">CC+"</span>))</span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>res5 <span class="ot">&lt;-</span> <span class="fu">get_model_result</span>(<span class="st">"Bnip3"</span>)</span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a>gg5 <span class="ot">&lt;-</span> <span class="fu">plot_response</span>(res5<span class="sc">$</span>m1, res5<span class="sc">$</span>m1_emm, res5<span class="sc">$</span>m1_pairs[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>),], <span class="at">palette =</span> <span class="st">"pal_okabe_ito_blue"</span>,</span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>                     <span class="at">x_axis_labels =</span> <span class="fu">c</span>(<span class="st">"DEX-</span><span class="sc">\n</span><span class="st">CC-"</span>, <span class="st">"DEX+</span><span class="sc">\n</span><span class="st">CC-"</span>, <span class="st">"DEX+</span><span class="sc">\n</span><span class="st">CC+"</span>, <span class="st">"DEX-</span><span class="sc">\n</span><span class="st">CC+"</span>))</span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(gg1, gg2, gg3, gg4, gg5, <span class="at">nrow =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="LSD1-inhibition-circumvents-glucocorticoid-induced-muscle-wasting-of-male-mice_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># gg1</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="co"># gg2</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co"># gg3</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co"># gg4</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co"># gg5</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>