<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jeff Walker">
<meta name="dcterms.date" content="2024-05-21">
<meta name="description" content="Fig 3g is a split-plot design. I often see researchers analyze these using separate paired t-tests. This isn’t so bad, but is limiting, and a better alternative is a linear mixed model. But does this replicate the Repeated Measures ANOVA in GraphPad used by the researchers? It does! Yaaay!">

<title>Data From - Fig 3g – The role of somatosensory innervation of adipose tissues</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="../../site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="../../site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Data From</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Fig 3g – The role of somatosensory innervation of adipose tissues</h1>
                  <div>
        <div class="description">
          Fig 3g is a split-plot design. I often see researchers analyze these using separate paired t-tests. This isn’t so bad, but is limiting, and a better alternative is a linear mixed model. But does this replicate the Repeated Measures ANOVA in GraphPad used by the researchers? It does! Yaaay!
        </div>
      </div>
                          <div class="quarto-categories">
                <div class="quarto-category">linear mixed model</div>
                <div class="quarto-category">random intercept</div>
                <div class="quarto-category">split plot</div>
                <div class="quarto-category">ggplot</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Jeff Walker </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">May 21, 2024</p>
      </div>
    </div>
    
      <div>
      <div class="quarto-title-meta-heading">Modified</div>
      <div class="quarto-title-meta-contents">
        <p class="date-modified">June 9, 2024</p>
      </div>
    </div>
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#vital-info" id="toc-vital-info" class="nav-link active" data-scroll-target="#vital-info">Vital info</a></li>
  <li><a href="#the-experiment" id="toc-the-experiment" class="nav-link" data-scroll-target="#the-experiment">The experiment</a></li>
  <li><a href="#what-is-a-randomized-split-plot-design-rspd" id="toc-what-is-a-randomized-split-plot-design-rspd" class="nav-link" data-scroll-target="#what-is-a-randomized-split-plot-design-rspd">What is a randomized split plot design (RSPD)?</a></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup">Setup</a></li>
  <li><a href="#import-and-wrangle" id="toc-import-and-wrangle" class="nav-link" data-scroll-target="#import-and-wrangle">Import and Wrangle</a></li>
  <li><a href="#fit-the-models" id="toc-fit-the-models" class="nav-link" data-scroll-target="#fit-the-models">Fit the models</a></li>
  <li><a href="#compare-lmm-to-rm-anova-and-to-graphpad-output" id="toc-compare-lmm-to-rm-anova-and-to-graphpad-output" class="nav-link" data-scroll-target="#compare-lmm-to-rm-anova-and-to-graphpad-output">Compare LMM to RM ANOVA and to GraphPad output</a>
  <ul class="collapse">
  <li><a href="#anova-replicates" id="toc-anova-replicates" class="nav-link" data-scroll-target="#anova-replicates">ANOVA replicates!</a></li>
  <li><a href="#contrasts-replicate" id="toc-contrasts-replicate" class="nav-link" data-scroll-target="#contrasts-replicate">Contrasts replicate!</a></li>
  </ul></li>
  <li><a href="#how-i-would-adjust-adjusting-all-6-responses-using-fdr" id="toc-how-i-would-adjust-adjusting-all-6-responses-using-fdr" class="nav-link" data-scroll-target="#how-i-would-adjust-adjusting-all-6-responses-using-fdr">How I would adjust: adjusting all 6 responses using FDR</a></li>
  <li><a href="#plotsreplicate" id="toc-plotsreplicate" class="nav-link" data-scroll-target="#plotsreplicate">Plots…replicate!</a></li>
  <li><a href="#published-methods-cut-and-pasted-from-the-article" id="toc-published-methods-cut-and-pasted-from-the-article" class="nav-link" data-scroll-target="#published-methods-cut-and-pasted-from-the-article">Published methods, cut and pasted from the article</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">





<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../../figs/The role of somatosensory innervation of adipose tissues/fig3g.png" class="img-fluid figure-img"></p>
<figcaption>Fig 3g from the article</figcaption>
</figure>
</div>
<section id="vital-info" class="level2">
<h2 class="anchored" data-anchor-id="vital-info">Vital info</h2>
<p>Data From: <a href="https://www.nature.com/articles/s41586-022-05137-7" target="_blank">Wang, Yu, et al.&nbsp;“The role of somatosensory innervation of adipose tissues.” Nature 609.7927 (2022): 569-574.</a></p>
<p>Fig: 3g <a href="../../data from/The role of somatosensory innervation of adipose tissues/41586_2022_5137_MOESM7_ESM.xlsx" target="_blank">download data</a></p>
<p>key words:</p>
<p>Published methods: repeated measures ANOVA</p>
<p>Design: Randomized Split Plot Design (RSPD)</p>
<p>Response: fold-change in gene expression</p>
<p>Key learning concepts: linear mixed model, random intercept, split-plot, repeated measures ANOVA</p>
<p>More info: <a href="https://www.middleprofessor.com/files/applied-biostatistics_bookdown/_book/lmm" target="_blank">Chapter 16 Models for non-independence – linear mixed models</a></p>
</section>
<section id="the-experiment" class="level2">
<h2 class="anchored" data-anchor-id="the-experiment">The experiment</h2>
<p>Treatments</p>
<p>Factor 1: tx</p>
<ol type="1">
<li>Saline (control). In the Excel file, this is labeled “SAL”. This should result in intact sympathetic activity.</li>
<li>6-OHDA (treatment). In the Excel file, this is labeled “OHDA”. This is 6-hydroxydopamine, a sympathetic neurotoxin that inhibits sympathatic activity.</li>
</ol>
<p>Factor 2: surgery</p>
<ol type="1">
<li>Cre- (control). In the Excel file, this is labeled “YFP”. This is a control for the Cre injection. This should result in an intact sensory system of fat.</li>
<li>Cre+ (treatment). In the Excel file, this is labeled “Cre”. This should result in an ablated sensory system of fat.</li>
</ol>
<p>Blocking or whole plot factor: ear_tag. This is the mouse ID.</p>
<p>For the tx treatment, either saline or 6-OHDA was delivered to both sides of a mouse. For the surgery treatment, one side was injected with Cre- and one with Cre+.</p>
<p>There are six response variables, each a log2 fold-change for the thermogenic and lipogenic genes:</p>
<ol type="1">
<li>Ucp1</li>
<li>Elovl3</li>
<li>Cidea</li>
<li>Fasn</li>
<li>Acacb</li>
<li>ChREBPβ</li>
</ol>
<p>comments:</p>
<ol type="1">
<li>Split plot design. Cre ablation of one limb under two different conditions. Each condition is a different mouse. Researchers used RMANOVA.</li>
<li>use base 2 for fold change! Yaaaay!</li>
<li>All statistical results in Excel file! Yaaaay! This should be mandatory.</li>
</ol>
</section>
<section id="what-is-a-randomized-split-plot-design-rspd" class="level2">
<h2 class="anchored" data-anchor-id="what-is-a-randomized-split-plot-design-rspd">What is a randomized split plot design (RSPD)?</h2>
<p>This is a randomized split plot design. tx (Sal or OHDA) is randomly assigned to Mouse - a mouse is the “plot”. Within a mouse, surgery (YFP or Cre) is assigned to side – this is the “subplot”. Mouse acts like a blocking factor for surgery but a completely randomized factor for tx. There is no subplot replication within a plot (mouse) so the linear mixed model (using R formula notation) is</p>
<p><code>fc ~ tx * surgery + (1 | ear_tag)</code></p>
<p>The factor ear_tag is a <strong>random intercept</strong>.</p>
<p>A question I had with this is, can I replicate the researcher’s results, as they used “Repeated Measures ANOVA” in GraphPad Prism. In general, I can replicate GraphPad results with a blocked design but I haven’t tried a split plot design (generally because I don’t think that I’ve found a paper with a split plot design that was analyzed in GraphPad using RM ANOVA)</p>
<p>Advantages of the linear mixed model</p>
<ol type="1">
<li>it can handle missing data – ANOVA throws out data points if there is any value missing value within the Plot variable (here, this is mouse ear_tag)</li>
<li>it is easy to handle non-normal data using generalized linear mixed models – ANOVA is for linear models only, although one can transform the response.</li>
<li>it is flexible, for example it is easy to add covariates. This isn’t so important for the bench biology data that I mostly see, although it is useful for example if a response was dependent on mouse weight and we wanted to adjust for weight without using a ratio.</li>
</ol>
<p>Advantages of the RM ANOVA model 1. It always works! LMM algorithms can fail, and not infrequently.</p>
</section>
<section id="setup" class="level2">
<h2 class="anchored" data-anchor-id="setup">Setup</h2>
</section>
<section id="import-and-wrangle" class="level2">
<h2 class="anchored" data-anchor-id="import-and-wrangle">Import and Wrangle</h2>
<p>Note: there are two sets of Fig 3g data. The bottom set looks like it is the top set recentered so that the mean of the SAL-YFP treatment has is mean zero. Since the data in the figure looked centered this way, I’ll import the bottom set.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">"The role of somatosensory innervation of adipose tissues"</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">"41586_2022_5137_MOESM7_ESM.xlsx"</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>fig3g <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">sheet =</span> <span class="st">"3g"</span>,</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">range =</span> <span class="st">"B34:K58"</span>,</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">col_names =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>() <span class="sc">|&gt;</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">clean_names</span>()</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>fig3g[, surgery <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(surgery, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"YFP"</span>, <span class="st">"Cre"</span>))]</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>fig3g[, tx <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(tx, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"SAL"</span>, <span class="st">"OHDA"</span>))]</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co"># output as clean excel file</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>fileout_name <span class="ot">&lt;-</span> <span class="st">"fig3g-SPD-The role of somatosensory innervation of adipose tissues.xlsx"</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>fileout_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, fileout_name)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">write_xlsx</span>(fig3g, fileout_path)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="fit-the-models" class="level2">
<h2 class="anchored" data-anchor-id="fit-the-models">Fit the models</h2>
<p>Create a function to fit the same models to each of the six responses. This is better practice than cut and paste because its less like to leave bugs if edits are made to code. Two models are fit: lmm1 – a linear mixed model with a random intercept, aov1 – a repeated measures ANOVA model</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>lmm1_fit <span class="ot">&lt;-</span> <span class="cf">function</span>(y_label){</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># y_label is the column name of the response variable</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  r_formula <span class="ot">&lt;-</span> <span class="fu">paste</span>(y_label, <span class="st">"~ tx * surgery + (1 | ear_tag)"</span>) <span class="sc">|&gt;</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">formula</span>()</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  lmm1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(r_formula, <span class="at">data =</span> fig3g)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(lmm1)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>aov1_fit <span class="ot">&lt;-</span> <span class="cf">function</span>(y_label){</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># y_label is the column name of the response variable</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  r_formula <span class="ot">&lt;-</span> <span class="fu">paste</span>(y_label, <span class="st">"~ tx * surgery + (surgery | ear_tag)"</span>) <span class="sc">|&gt;</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">formula</span>()</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(r_formula, <span class="at">data =</span> fig3g)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(aov1)</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>fig3g_lmm1 <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>fig3g_aov1 <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>y_list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ucp1"</span>, <span class="st">"elovl3"</span>, <span class="st">"cidea"</span>, <span class="st">"fasn"</span>, <span class="st">"acacb"</span>, <span class="st">"ch_reb_pb"</span>)</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(y_list)){</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  y_label <span class="ot">&lt;-</span> y_list[i]</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>  fig3g_lmm1[[y_label]] <span class="ot">&lt;-</span> <span class="fu">lmm1_fit</span>(y_label)</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>  fig3g_aov1[[y_label]] <span class="ot">&lt;-</span> <span class="fu">aov1_fit</span>(y_label)</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a><span class="co"># get emmeans using function</span></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>emmeans_out <span class="ot">&lt;-</span> <span class="cf">function</span>(m1){</span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># m1 is the fit model</span></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>  m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(m1,</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>                    <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"tx"</span>, <span class="st">"surgery"</span>))</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(m1_emm)</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>fig3g_lmm1_emm <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>fig3g_aov1_emm <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(y_list)){</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>  y_label <span class="ot">&lt;-</span> y_list[i]</span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>  fig3g_lmm1_emm[[y_label]] <span class="ot">&lt;-</span> <span class="fu">emmeans_out</span>(fig3g_lmm1[[y_label]])</span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>  fig3g_aov1_emm[[y_label]] <span class="ot">&lt;-</span> <span class="fu">emmeans_out</span>(fig3g_aov1[[y_label]])</span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>pairs_out <span class="ot">&lt;-</span> <span class="cf">function</span>(m1_emm){</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># m1 is the fit model</span></span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>  m1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(m1_emm,</span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">"revpairwise"</span>,</span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a>                       <span class="at">simple =</span> <span class="st">"each"</span>,</span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>                       <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">"none"</span>) <span class="sc">|&gt;</span></span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(m1_pairs)</span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a>fig3g_lmm1_pairs <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>fig3g_aov1_pairs <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(y_list)){</span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a>  y_label <span class="ot">&lt;-</span> y_list[i]</span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a>  fig3g_lmm1_pairs[[y_label]] <span class="ot">&lt;-</span> <span class="fu">pairs_out</span>(fig3g_lmm1_emm[[y_label]])</span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>  fig3g_aov1_pairs[[y_label]] <span class="ot">&lt;-</span> <span class="fu">pairs_out</span>(fig3g_aov1_emm[[y_label]])</span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="compare-lmm-to-rm-anova-and-to-graphpad-output" class="level2">
<h2 class="anchored" data-anchor-id="compare-lmm-to-rm-anova-and-to-graphpad-output">Compare LMM to RM ANOVA and to GraphPad output</h2>
<section id="anova-replicates" class="level3">
<h3 class="anchored" data-anchor-id="anova-replicates">ANOVA replicates!</h3>
<p>tested using ucp1 only</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(fig3g_lmm1[[<span class="st">"ucp1"</span>]]) <span class="sc">|&gt;</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">4</span>,</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">caption =</span> <span class="st">"LMM fit"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<table class="table table-sm table-striped small" data-quarto-postprocess="true">
<caption>LMM fit</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Sum Sq</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Mean Sq</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">NumDF</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">DenDF</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">F value</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Pr(&gt;F)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">tx</td>
<td style="text-align: right;">24.8575</td>
<td style="text-align: right;">24.8575</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">8.3181</td>
<td style="text-align: right;">0.0163</td>
</tr>
<tr class="even">
<td style="text-align: left;">surgery</td>
<td style="text-align: right;">31.4641</td>
<td style="text-align: right;">31.4641</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">10.5288</td>
<td style="text-align: right;">0.0088</td>
</tr>
<tr class="odd">
<td style="text-align: left;">tx:surgery</td>
<td style="text-align: right;">5.2770</td>
<td style="text-align: right;">5.2770</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">1.7658</td>
<td style="text-align: right;">0.2134</td>
</tr>
</tbody>
</table>


</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(fig3g_aov1[[<span class="st">"ucp1"</span>]]) <span class="sc">|&gt;</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">4</span>, <span class="at">caption =</span> <span class="st">"RM ANOVA fit"</span>) <span class="sc">|&gt;</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<table class="table table-sm table-striped small" data-quarto-postprocess="true">
<caption>RM ANOVA fit</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">num Df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">den Df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">MSE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">F</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">ges</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Pr(&gt;F)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">tx</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">6.2148</td>
<td style="text-align: right;">8.3181</td>
<td style="text-align: right;">0.3597</td>
<td style="text-align: right;">0.0163</td>
</tr>
<tr class="even">
<td style="text-align: left;">surgery</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">2.9884</td>
<td style="text-align: right;">10.5288</td>
<td style="text-align: right;">0.2548</td>
<td style="text-align: right;">0.0088</td>
</tr>
<tr class="odd">
<td style="text-align: left;">tx:surgery</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">2.9884</td>
<td style="text-align: right;">1.7658</td>
<td style="text-align: right;">0.0542</td>
<td style="text-align: right;">0.2134</td>
</tr>
</tbody>
</table>


</div>
</div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../../figs/The role of somatosensory innervation of adipose tissues/fig3g_ucp1_anova_table.png" class="img-fluid figure-img"></p>
<figcaption>Fig 3g ANOVA table from Excel file</figcaption>
</figure>
</div>
<p>lmm1 and aov1 give the same results here <strong>because there is no missing data</strong>, which the lmm is better at handling. Both replicate the GraphPad Prism results.</p>
</section>
<section id="contrasts-replicate" class="level3">
<h3 class="anchored" data-anchor-id="contrasts-replicate">Contrasts replicate!</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create a column of sidak adjusted p-values in the contrast tables</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(y_list)){ <span class="co"># over the 6 responses</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  m1_pairs <span class="ot">&lt;-</span> fig3g_lmm1_pairs[[y_list[i]]]</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  m1_pairs[<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>, p_sidak <span class="sc">:</span><span class="er">=</span> <span class="fu">Sidak.p.adjust</span>(p.value)]</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  fig3g_lmm1_pairs[[y_list[i]]] <span class="ot">&lt;-</span> m1_pairs</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co"># make a pretty output table</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>contrast_table <span class="ot">&lt;-</span> <span class="fu">data.table</span>(<span class="cn">NULL</span>)</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>keep_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"tx"</span>, <span class="st">"contrast"</span>, <span class="st">"estimate"</span>, <span class="st">"p_sidak"</span>)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(y_list)){ <span class="co"># over the 6 responses</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>  m1_pairs <span class="ot">&lt;-</span> fig3g_lmm1_pairs[[y_list[i]]]</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  contrast_table <span class="ot">&lt;-</span> <span class="fu">rbind</span>(contrast_table,</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>                          m1_pairs[<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>, .SD, <span class="at">.SDcols =</span> keep_cols])</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>contrast_table <span class="sc">|&gt;</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">4</span>) <span class="sc">|&gt;</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">|&gt;</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pack_rows</span>(y_list[<span class="dv">1</span>], <span class="dv">1</span>, <span class="dv">2</span>) <span class="sc">|&gt;</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pack_rows</span>(y_list[<span class="dv">2</span>], <span class="dv">3</span>, <span class="dv">4</span>) <span class="sc">|&gt;</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pack_rows</span>(y_list[<span class="dv">3</span>], <span class="dv">5</span>, <span class="dv">6</span>) <span class="sc">|&gt;</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pack_rows</span>(y_list[<span class="dv">4</span>], <span class="dv">7</span>, <span class="dv">8</span>) <span class="sc">|&gt;</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pack_rows</span>(y_list[<span class="dv">5</span>], <span class="dv">9</span>, <span class="dv">10</span>) <span class="sc">|&gt;</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pack_rows</span>(y_list[<span class="dv">6</span>], <span class="dv">11</span>, <span class="dv">12</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<table class="table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">tx</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p_sidak</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="2">
<td colspan="4" style="border-bottom: 1px solid"><strong>ucp1</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">SAL</td>
<td style="text-align: left;">Cre - YFP</td>
<td style="text-align: right;">-3.2278</td>
<td style="text-align: right;">0.0178</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">OHDA</td>
<td style="text-align: left;">Cre - YFP</td>
<td style="text-align: right;">-1.3522</td>
<td style="text-align: right;">0.3684</td>
</tr>
<tr class="even" data-grouplength="2">
<td colspan="4" style="border-bottom: 1px solid"><strong>elovl3</strong></td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">SAL</td>
<td style="text-align: left;">Cre - YFP</td>
<td style="text-align: right;">-3.5792</td>
<td style="text-align: right;">0.0083</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">OHDA</td>
<td style="text-align: left;">Cre - YFP</td>
<td style="text-align: right;">-1.5600</td>
<td style="text-align: right;">0.2574</td>
</tr>
<tr class="odd" data-grouplength="2">
<td colspan="4" style="border-bottom: 1px solid"><strong>cidea</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">SAL</td>
<td style="text-align: left;">Cre - YFP</td>
<td style="text-align: right;">-2.2268</td>
<td style="text-align: right;">0.0119</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">OHDA</td>
<td style="text-align: left;">Cre - YFP</td>
<td style="text-align: right;">-0.8875</td>
<td style="text-align: right;">0.3538</td>
</tr>
<tr class="even" data-grouplength="2">
<td colspan="4" style="border-bottom: 1px solid"><strong>fasn</strong></td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">SAL</td>
<td style="text-align: left;">Cre - YFP</td>
<td style="text-align: right;">-1.8012</td>
<td style="text-align: right;">0.0060</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">OHDA</td>
<td style="text-align: left;">Cre - YFP</td>
<td style="text-align: right;">-0.6543</td>
<td style="text-align: right;">0.3409</td>
</tr>
<tr class="odd" data-grouplength="2">
<td colspan="4" style="border-bottom: 1px solid"><strong>acacb</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">SAL</td>
<td style="text-align: left;">Cre - YFP</td>
<td style="text-align: right;">-1.8589</td>
<td style="text-align: right;">0.0147</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">OHDA</td>
<td style="text-align: left;">Cre - YFP</td>
<td style="text-align: right;">-0.4956</td>
<td style="text-align: right;">0.6315</td>
</tr>
<tr class="even" data-grouplength="2">
<td colspan="4" style="border-bottom: 1px solid"><strong>ch_reb_pb</strong></td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">SAL</td>
<td style="text-align: left;">Cre - YFP</td>
<td style="text-align: right;">-2.7360</td>
<td style="text-align: right;">0.0069</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">OHDA</td>
<td style="text-align: left;">Cre - YFP</td>
<td style="text-align: right;">-1.3383</td>
<td style="text-align: right;">0.1767</td>
</tr>
</tbody>
</table>


</div>
</div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../../figs/The role of somatosensory innervation of adipose tissues/fig3g_contrasts.png" class="img-fluid figure-img"></p>
<figcaption>Fig 3g contrasts from the Excel file</figcaption>
</figure>
</div>
<p>Notes</p>
<ol type="1">
<li>the direction of my contrast is Cre+ - Cre- (or, using the Excel file labels, this is Cre - YFP), since I like the direction to be what happens when you add the treatment, relative to a control.</li>
<li>The authors used a Sidak adjustment on the two simple Cre- - Cre+ contrasts only.</li>
</ol>
<p>The contrasts replicate (other than the sign, but that is because I chose the opposite direction of the contrast)!</p>
<p>I do have thoughts on the correction:</p>
<ol type="1">
<li>The researchers do not state why the set of adjusted p-values is 2 (the two simple effects of surgery) and not 4 (the additional two simple effects tx).</li>
<li>The holm method of adjustment is more powerful than Sidak.</li>
<li>I would <em>not</em> adjust if I were only investigating ucp1 as the response. Each of these is a test of its own question with different expectations given working knowledge. See xxx.</li>
<li>That said, there are 6 response variables from the experiment and if our hypothesis is something like “is there some difference in gene expression given this set of genes?” then I would adjust using the FDR, as this is a question applied to the whole gene set. This is exactly the norm in gene expression studies with thousands of genes. I never see researchers adjust for multiple responses except in these large large gene expression studies (or similar). So there is inconsistency in the logic applied to the statistical analysis not just within a field but within a single paper.</li>
</ol>
</section>
</section>
<section id="how-i-would-adjust-adjusting-all-6-responses-using-fdr" class="level2">
<h2 class="anchored" data-anchor-id="how-i-would-adjust-adjusting-all-6-responses-using-fdr">How I would adjust: adjusting all 6 responses using FDR</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this seems ultra klunky but it works</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="fu">length</span>(y_list))</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>){ <span class="co"># do this for each simple effect in the contrast table</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(y_list)){ <span class="co"># over the 6 responses</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>    m1_pairs <span class="ot">&lt;-</span> fig3g_lmm1_pairs[[y_list[i]]]</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>    p[i] <span class="ot">&lt;-</span> m1_pairs[j, p.value]</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  p_adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(p, <span class="st">"fdr"</span>)</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(y_list)){ <span class="co"># over the 6 responses</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>    m1_pairs <span class="ot">&lt;-</span> fig3g_lmm1_pairs[[y_list[i]]]</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>    m1_pairs[j, p_fdr <span class="sc">:</span><span class="er">=</span> p_adj[i]]</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>    fig3g_lmm1_pairs[[y_list[i]]] <span class="ot">&lt;-</span> m1_pairs</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="co"># create a table of p-values for cre contrasts only</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>out_table <span class="ot">&lt;-</span> <span class="fu">data.table</span>(</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">response =</span> y_list,</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">Sal_p =</span> <span class="fu">numeric</span>(<span class="fu">length</span>(y_list)),</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">OHDA_p =</span> <span class="fu">numeric</span>(<span class="fu">length</span>(y_list)),</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">Sal_fdr =</span> <span class="fu">numeric</span>(<span class="fu">length</span>(y_list)),</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">OHDA_fdr =</span> <span class="fu">numeric</span>(<span class="fu">length</span>(y_list))</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(y_list)){ <span class="co"># over the 6 responses</span></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>    m1_pairs <span class="ot">&lt;-</span> fig3g_lmm1_pairs[[y_list[i]]]</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>    out_table[response <span class="sc">==</span> y_list[[i]], Sal_p <span class="sc">:</span><span class="er">=</span> m1_pairs[<span class="dv">3</span>, p.value]]</span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>    out_table[response <span class="sc">==</span> y_list[[i]], OHDA_p<span class="sc">:</span><span class="er">=</span> m1_pairs[<span class="dv">4</span>, p.value]]</span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>    out_table[response <span class="sc">==</span> y_list[[i]], Sal_fdr <span class="sc">:</span><span class="er">=</span> m1_pairs[<span class="dv">3</span>, p_fdr]]</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>    out_table[response <span class="sc">==</span> y_list[[i]], OHDA_fdr <span class="sc">:</span><span class="er">=</span> m1_pairs[<span class="dv">4</span>, p_fdr]]</span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>out_table <span class="sc">|&gt;</span></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">caption =</span> <span class="st">"unadjusted and FDR adjust p-values for Cre+ - Cre- contrasts within Saline and 6-OHDA treatments"</span>) <span class="sc">|&gt;</span></span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<table class="table table-sm table-striped small" data-quarto-postprocess="true">
<caption>unadjusted and FDR adjust p-values for Cre+ - Cre- contrasts within Saline and 6-OHDA treatments</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">response</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Sal_p</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">OHDA_p</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Sal_fdr</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">OHDA_fdr</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ucp1</td>
<td style="text-align: right;">0.0089590</td>
<td style="text-align: right;">0.2052979</td>
<td style="text-align: right;">0.0089590</td>
<td style="text-align: right;">0.2463575</td>
</tr>
<tr class="even">
<td style="text-align: left;">elovl3</td>
<td style="text-align: right;">0.0041345</td>
<td style="text-align: right;">0.1382491</td>
<td style="text-align: right;">0.0082690</td>
<td style="text-align: right;">0.2463575</td>
</tr>
<tr class="odd">
<td style="text-align: left;">cidea</td>
<td style="text-align: right;">0.0059646</td>
<td style="text-align: right;">0.1961155</td>
<td style="text-align: right;">0.0088577</td>
<td style="text-align: right;">0.2463575</td>
</tr>
<tr class="even">
<td style="text-align: left;">fasn</td>
<td style="text-align: right;">0.0030145</td>
<td style="text-align: right;">0.1881198</td>
<td style="text-align: right;">0.0082690</td>
<td style="text-align: right;">0.2463575</td>
</tr>
<tr class="odd">
<td style="text-align: left;">acacb</td>
<td style="text-align: right;">0.0073814</td>
<td style="text-align: right;">0.3929358</td>
<td style="text-align: right;">0.0088577</td>
<td style="text-align: right;">0.3929358</td>
</tr>
<tr class="even">
<td style="text-align: left;">ch_reb_pb</td>
<td style="text-align: right;">0.0034794</td>
<td style="text-align: right;">0.0926347</td>
<td style="text-align: right;">0.0082690</td>
<td style="text-align: right;">0.2463575</td>
</tr>
</tbody>
</table>


</div>
</div>
</div>
</section>
<section id="plotsreplicate" class="level2">
<h2 class="anchored" data-anchor-id="plotsreplicate">Plots…replicate!</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>fig3g[, cre <span class="sc">:</span><span class="er">=</span> <span class="fu">ifelse</span>(surgery <span class="sc">==</span> <span class="st">"YFP"</span>, <span class="st">"Cre-"</span>, <span class="st">"Cre+"</span>)]</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>fig3g[, ohda <span class="sc">:</span><span class="er">=</span> <span class="fu">ifelse</span>(tx <span class="sc">==</span> <span class="st">"SAL"</span>, <span class="st">"SAL"</span>, <span class="st">"6-OHDA"</span>)]</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>fig3g[, treatment <span class="sc">:</span><span class="er">=</span> <span class="fu">paste</span>(ohda, cre, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>) <span class="sc">|&gt;</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>      <span class="fu">factor</span>(<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"SAL</span><span class="sc">\n</span><span class="st">Cre-"</span>, <span class="st">"SAL</span><span class="sc">\n</span><span class="st">Cre+"</span>, <span class="st">"6-OHDA</span><span class="sc">\n</span><span class="st">Cre-"</span>, <span class="st">"6-OHDA</span><span class="sc">\n</span><span class="st">Cre+"</span>))]</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>plot_it <span class="ot">&lt;-</span> <span class="cf">function</span>(y_col,</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>                    m1_pairs){</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  gg <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> fig3g,</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>               <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>                   <span class="at">y =</span> <span class="fu">get</span>(y_col))) <span class="sc">+</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> surgery),</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>               <span class="at">size =</span> <span class="dv">2</span>,</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>               <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> ear_tag),</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">"Log 2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(y_col) <span class="sc">+</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal_okabe_ito_2) <span class="sc">+</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_pubr</span>() <span class="sc">+</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add p-values</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>  p_table <span class="ot">&lt;-</span> <span class="fu">data.table</span>(</span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">group1 =</span> <span class="fu">c</span>(<span class="st">"SAL</span><span class="sc">\n</span><span class="st">Cre-"</span>, <span class="st">"6-OHDA</span><span class="sc">\n</span><span class="st">Cre-"</span>),</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">group2 =</span> <span class="fu">c</span>(<span class="st">"SAL</span><span class="sc">\n</span><span class="st">Cre+"</span>, <span class="st">"6-OHDA</span><span class="sc">\n</span><span class="st">Cre+"</span>),</span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">p =</span> m1_pairs[<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>, p_sidak] <span class="sc">|&gt;</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>      <span class="fu">p_round</span>(<span class="at">digits =</span> <span class="dv">4</span>) <span class="sc">|&gt;</span> <span class="co">#rstatix package</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>      <span class="fu">p_format</span>(<span class="at">digits =</span> <span class="dv">4</span>, <span class="co">#rstatix package</span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>                        <span class="at">accuracy =</span> <span class="fl">1e-04</span>,</span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>                        <span class="at">add.p =</span> <span class="cn">TRUE</span>),</span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">y.position =</span> fig3g[, <span class="fu">max</span>(<span class="fu">get</span>(y_col))] <span class="sc">+</span> </span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>      <span class="fl">0.05</span><span class="sc">*</span>(fig3g[, <span class="fu">max</span>(<span class="fu">get</span>(y_col))] <span class="sc">-</span> fig3g[, <span class="fu">min</span>(<span class="fu">get</span>(y_col))])</span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>  gg <span class="ot">&lt;-</span> gg <span class="sc">+</span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_pvalue_manual</span>(</span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a>      <span class="at">data =</span> p_table,</span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>      <span class="at">tip.length =</span> <span class="fl">0.001</span>)</span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(gg)</span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a>gg_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(y_list)){</span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a>  m1_pairs <span class="ot">&lt;-</span> fig3g_lmm1_pairs[[y_list[i]]]</span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a>  <span class="co"># use the sidak p-values</span></span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a>  m1_pairs[, p.value <span class="sc">:</span><span class="er">=</span> p_sidak]</span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a>  gg <span class="ot">&lt;-</span> <span class="fu">plot_it</span>(<span class="at">y_col =</span> y_list[i], m1_pairs)</span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a>  gg_list[[y_list[i]]] <span class="ot">&lt;-</span> gg</span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> gg_list, <span class="at">nrow =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="The-role-of-somatosensory-innervation-of-adipose-tissues_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="1056"></p>
</figure>
</div>
</div>
</div>
<p>What’s going on with the direction of the effect in Fig 3g? Compare my figure with the published figure.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../../figs/The role of somatosensory innervation of adipose tissues/fig3g.png" class="img-fluid figure-img"></p>
<figcaption>Fig 3g</figcaption>
</figure>
</div>
<p>Compared to my figure built from the archived data, the data in Fig3g is reflected about the horizontal axis at y = 0, that is, the data are upside down, so the apparent direction of the differences is exactly opposite that of the archived data in the Excel file. The GraphPad prism contrasts tables in the Excel file match the archived data in the Excel file and not the plotted data. This doesn’t matter for the p-values but since directions of the differences are exactly opposite, the interpretation of the results is exactly opposite.</p>
<p>So, what’s up with the reflected (upside down) data? The matrices of archived values are <span class="math inline">\(\Delta C_q\)</span> and centered (<span class="math inline">\(\Delta\Delta C_q\)</span>) values, which decrease as expression levels increase. We want an increase in expression to look like an increase in the y-value. So to get the effects in this meaningful direction, we can multiply the <span class="math inline">\(\Delta C_q\)</span> values by -1.</p>
<p>Let’s reflect the <span class="math inline">\(\Delta\Delta C_q\)</span> values by multiplying them by -1 and replot:</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># reflect the values in the response variables</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>fig3g[, ucp1_ref <span class="sc">:</span><span class="er">=</span> <span class="sc">-</span>ucp1]</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>fig3g[, elovl3_ref <span class="sc">:</span><span class="er">=</span> <span class="sc">-</span>elovl3]</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>fig3g[, cidea_ref <span class="sc">:</span><span class="er">=</span> <span class="sc">-</span>cidea]</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>fig3g[, fasn_ref <span class="sc">:</span><span class="er">=</span> <span class="sc">-</span>fasn]</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>fig3g[, acacb_ref <span class="sc">:</span><span class="er">=</span> <span class="sc">-</span>acacb]</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>fig3g[, ch_reb_pb_ref <span class="sc">:</span><span class="er">=</span> <span class="sc">-</span>ch_reb_pb]</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>y_list_ref <span class="ot">&lt;-</span> <span class="fu">paste0</span>(y_list, <span class="st">"_ref"</span>)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>gg_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(y_list)){</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  m1_pairs <span class="ot">&lt;-</span> fig3g_lmm1_pairs[[y_list[i]]]</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># use the sidak p-values</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>  m1_pairs[, p.value <span class="sc">:</span><span class="er">=</span> p_sidak]</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>  gg <span class="ot">&lt;-</span> <span class="fu">plot_it</span>(<span class="at">y_col =</span> y_list_ref[i], m1_pairs)</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>  gg_list[[y_list[i]]] <span class="ot">&lt;-</span> gg</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> gg_list, <span class="at">nrow =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="The-role-of-somatosensory-innervation-of-adipose-tissues_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="1056"></p>
</figure>
</div>
</div>
</div>
<p>Okay - this replicates the plot!</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(y_list)){</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  m1 <span class="ot">&lt;-</span> fig3g_lmm1[[y_list[i]]]</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="published-methods-cut-and-pasted-from-the-article" class="level2">
<h2 class="anchored" data-anchor-id="published-methods-cut-and-pasted-from-the-article">Published methods, cut and pasted from the article</h2>
<p>A combination of axonal target injection of retrograde Cre and somatic expression of a Cre-dependent payload has been widely used to manipulate projection-specific circuits in the brain. However, legacy peripheral viral tracers such as pseudorabies virus and herpes simplex virus are highly toxic, restricting their use beyond acute anatomical mapping. In search for newer and safer viral vectors suitable for long-term functional manipulations, we found that AAV9 exhibited a high retrograde potential from iWAT to DRGs (Extended Data Fig. 4a). We adopted a published viral engineering pipeline22 to generate randomized mutants of AAV9 (Extended Data Fig. 4b,c). Although our initial intent was to improve retrograde efficiency from fat to DRGs, the evolved new retrograde vector optimized for organ tracing (or ROOT) is more desirable mainly for its significantly reduced off-target expression such as in SChGs, contralateral DRGs and the liver (Fig. 2a,b and Extended Data Fig. 4d–g).</p>
<p>ROOT provides an opportunity to specifically ablate the sensory innervation in fat—we injected Cre-dependent diphtheria toxin subunit A (DTA) construct (mCherry-flex-DTA) into the T13/L1 DRGs bilaterally while injecting Cre- or YFP-expressing ROOT unilaterally in the iWATs (Fig. 2c).</p>
<p>As thermogenic and lipogenesis programs are both downstream of sympathetic signalling24,26,29, we next tested whether the sensory-elicited gene expression changes are dependent on intact sympathetic innervation. We bilaterally injected 6-OHDA—a catecholaminergic toxin for selective sympathetic denervation (Extended Data Fig. 6c,d)—into the iWAT of mice that had previously undergone unilateral sensory ablation (Fig. 3f).</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>